### ISSUE TITLE
Cannot specify TZEndpoints without specifying HostSwitchMode at HostSwitch

### ISSUE THREAD

**Issue by Dyanngg** [vmware/ansible-for-nsxt#219]
With PR #214, support is dropped for `transport_zone_endpoints` on transport node level and only considering `transport_zone_endpoints` in host_switch_spec level. This has several pitfalls:
1. It seems nsx is not accepting transport_zone_enpoints in host_switch_spec unless  HostSwitchMode is also specified, even though on API it only states the other way around:
```
            "display_name": "edge-transp-node-1",
            "host_switch_spec": {
                "host_switches": [
                    {
                        "host_switch_name": "hostswitch-overlay",
                        "host_switch_profile_ids": [
                            {
                                "key": "UplinkHostSwitchProfile",
                                "value": "675a4e00-d4be-4acf-8edf-f44fbcb8f371"
                            }
                        ],
                        "ip_assignment_spec": {
                            "ip_pool_id": "f3452c42-7401-434c-886f-7ebd7bc3e1ec",
                            "resource_type": "StaticIpPoolSpec"
                        },
                        "pnics": [
                            {
                                "device_name": "fp-eth0",
                                "uplink_name": "uplink-1"
                            }
                        ],
                        "transport_zone_endpoints": [
                            {
                                "transport_zone_id": "40d20e00-7af1-460c-8ff8-97b9c384bb2f"
                            },
                            {
                                "transport_zone_id": "96c557d5-1469-49ac-a3ec-ede9824125b2"
                            }
                        ]
                    }
                ],
                "resource_type": "StandardHostSwitchSpec"
            }
```
With this spec, the following error will rise:
``Error[(9542, {u'error_code': 9542, u'error_message': u'Cannot specify TransportZoneEndpoints without specifying HostSwitchMode at HostSwitch hostswitch-overlay.', u'httpStatus': u'BAD_REQUEST', u'module_name': u'NsxSwitching service'})]."``

2. ``dev`` branch is still used by NSXT2.5 deployments in [nsx-t concourse pipeline ](https://github.com/vmware/nsx-t-datacenter-ci-pipelines), which is incompatible with this change. Please update the branch accordingly to unblock 2.5.0 usage.

**Comment by Akhilesh2412** [comment]
Hello @Dyanngg 
The support is not dropped. But, we got see why it is failing.

**Comment by Akhilesh2412** [comment]
Hello Madhu
Please verify if the request being sent is valid one? Because the error is being thrown by the nsx manager server saying BAD_REQUEST.

**Comment by madhukark** [comment]
The request needs '"host_switch_mode": "STANDARD"' inside host_switches. Can you please try after adding it.

The API doc mentions this:
host_switch_mode: 
This mode is available only on ESX hypervisor (7.0 and above).
If this property is specified, transport_zone_endpoints must be specified at StandardHostSwitch level.

transport_zone_endpoints:
If this property is specified, host_switch_mode at StandardHostSwitch level must be specified.

**Comment by madhukark** [comment]
@Akhilesh2412  please make sure the module works for input specified in 2.5 format (transport_zone_endpoint outside host_switch_spec)
